sdk/apis/cache/v1alpha1/types_cachedresources.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2024 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// CachedResource defines a resource that is cached in the cache.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+type CachedResource struct {
```

Comment:
PublishedObject ? This is neither a resource, nor being cached is unique (there are other cache objects).

Reply:
So the main object the user interacts with is `PublishedResources`, naming this `PublishedObject` kinda creates (in my head) this question - what is the difference? Names are very confusing, 

I agree `CachedResource` is not ideal, but I'm not sure what a better name is?

Reply:
Imho `PublishedObject` and `CachedObject` would be fine.

The relation of a `PublishedResource` turning into one or many `PublishedObject`/`PublishedObjects` would be pretty clear.

`CachedObject` would work as well I think and leave it open to be used from other sources.